include/zephyr/dt-bindings/clock/stm32u5_clock.h

```diff
@@ -21,4 +50,73 @@
 #define STM32_PERIPH_BUS_MIN	STM32_CLOCK_BUS_AHB1
 #define STM32_PERIPH_BUS_MAX	STM32_CLOCK_BUS_APB3
 
+/**
+ * @brief STM32U5 clock configuration bit field.
+ *
+ * - reg   (1/2/3)         [ 0 : 7 ]
+ * - shift (0..31)         [ 8 : 12 ]
+ * - mask  (0x1, 0x3, 0x7) [ 13 : 15 ]
+ * - val   (0..7)          [ 16 : 18 ]
+ *
+ * @param reg RCC_CCIPRx register offset
+ * @param shift Position within RCC_CCIPRx.
+ * @param mask Mask for the RCC_CCIPRx field.
+ * @param val Clock value (0, 1, ... 7).
+ */
+
+#define STM32_CLOCK_REG_MASK    0xFFU
+#define STM32_CLOCK_REG_SHIFT   0U
+#define STM32_CLOCK_SHIFT_MASK  0x1FU
+#define STM32_CLOCK_SHIFT_SHIFT 8U
+#define STM32_CLOCK_MASK_MASK   0x7U
+#define STM32_CLOCK_MASK_SHIFT  13U
+#define STM32_CLOCK_VAL_MASK    0x7U
+#define STM32_CLOCK_VAL_SHIFT   16U
+
+#define STM32_CLOCK(val, mask, shift, reg)					\
+	((((reg) & STM32_CLOCK_REG_MASK) << STM32_CLOCK_REG_SHIFT) |	\
+	 (((shift) & STM32_CLOCK_SHIFT_MASK) << STM32_CLOCK_SHIFT_SHIFT) |	\
+	 (((mask) & STM32_CLOCK_MASK_MASK) << STM32_CLOCK_MASK_SHIFT) |	\
+	 (((val) & STM32_CLOCK_VAL_MASK) << STM32_CLOCK_VAL_SHIFT))
+
+/** @brief RCC_CCIPRx register offset (RM0456.pdf) */
+#define CCIPR1_REG		0xE0
+#define CCIPR2_REG		0xE4
+#define CCIPR3_REG		0xE8
+
+/** @brief Device clk sources selection helpers (RM0399.pdf) */
+/** CCIPR1 devices */
+#define USART1_SEL(val)		STM32_CLOCK(val, 3, 0, CCIPR1_REG)
+#define USART2_SEL(val)		STM32_CLOCK(val, 3, 2, CCIPR1_REG)
+#define USART3_SEL(val)		STM32_CLOCK(val, 3, 4, CCIPR1_REG)
+#define USART4_SEL(val)		STM32_CLOCK(val, 3, 6, CCIPR1_REG)
+#define USART5_SEL(val)		STM32_CLOCK(val, 3, 8, CCIPR1_REG)
+#define I2C1_SEL(val)		STM32_CLOCK(val, 3, 10, CCIPR1_REG)
+#define I2C2_SEL(val)		STM32_CLOCK(val, 3, 12, CCIPR1_REG)
+#define I2C4_SEL(val)		STM32_CLOCK(val, 3, 14, CCIPR1_REG)
+#define SPI2_SEL(val)		STM32_CLOCK(val, 3, 16, CCIPR1_REG)
+#define LPTIM2_SEL(val)		STM32_CLOCK(val, 3, 18, CCIPR1_REG)
+#define SPI1_SEL(val)		STM32_CLOCK(val, 3, 20, CCIPR1_REG)
+#define SYSTICK_SEL(val)	STM32_CLOCK(val, 3, 22, CCIPR1_REG)
+#define FDCAN1_SEL(val)		STM32_CLOCK(val, 3, 24, CCIPR1_REG)
+#define ICKLK_SEL(val)		STM32_CLOCK(val, 3, 26, CCIPR1_REG)
+#define TIMIC_SEL(val)		STM32_CLOCK(val, 7, 29, CCIPR1_REG)
+/** CCIPR2 devices */
+#define MDF1_SEL(val)		STM32_CLOCK(val, 7, 0, CCIPR2_REG)
+#define SAI1_SEL(val)		STM32_CLOCK(val, 7, 5, CCIPR2_REG)
+#define SAI2_SEL(val)		STM32_CLOCK(val, 7, 8, CCIPR2_REG)
+#define SAE_SEL(val)		STM32_CLOCK(val, 1, 11, CCIPR2_REG)
+#define RNG_SEL(val)		STM32_CLOCK(val, 3, 12, CCIPR2_REG)
+#define SDMMC_SEL(val)		STM32_CLOCK(val, 1, 14, CCIPR2_REG)
+#define OCTOSPI_SEL(val)	STM32_CLOCK(val, 3, 20, CCIPR2_REG)
+/** CCIPR3 devices */
+
 #endif /* ZEPHYR_INCLUDE_DT_BINDINGS_CLOCK_STM32U5_CLOCK_H_ */
```

Comment:
Shall be moved to the end of the file, otherwise selectors of CCIPR3_REG are redefined.